Serialization issue when doing Property.of() on plugin-ldap repo

### Describe the issue

During migration to Property v2 for plugin-ldap ([PR](https://github.com/kestra-io/plugin-ldap/pull/23))
I encounter an issue when trying to use `Property.of(SearchScope.SUB)`. 

Here is the problematic line : https://github.com/kestra-io/plugin-ldap/blob/17eed04721bceed853b0c24ec8a892c296f9fdf7/src/main/java/io/kestra/plugin/ldap/Search.java#L103

Here is the stack trace associated : 
```
Caused by: java.lang.IllegalArgumentException: Cannot deserialize value of type `java.lang.String` from Object value (token `JsonToken.START_OBJECT`)
     at [Source: UNKNOWN; byte offset: #UNKNOWN]
    	at com.fasterxml.jackson.databind.ObjectMapper._convert(ObjectMapper.java:4636)
    	at com.fasterxml.jackson.databind.ObjectMapper.convertValue(ObjectMapper.java:4567)
    	at io.kestra.core.models.property.Property.of(Property.java:65)
    	at io.kestra.plugin.ldap.Search.$default$sub(Search.java:104)
    	at io.kestra.plugin.ldap.Search.<init>(Search.java:27)
    	at java.base/jdk.internal.reflect.DirectConstructorHandleAccessor.newInstance(DirectConstructorHandleAccessor.java:62)
    	at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:502)
    	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:486)
    	at java.base/java.util.ServiceLoader$ProviderImpl.newInstance(ServiceLoader.java:789)
    	... 165 common frames omitted
    Caused by: com.fasterxml.jackson.databind.exc.MismatchedInputException: Cannot deserialize value of type `java.lang.String` from Object value (token `JsonToken.START_OBJECT`)
     at [Source: UNKNOWN; byte offset: #UNKNOWN]
    	at com.fasterxml.jackson.databind.exc.MismatchedInputException.from(MismatchedInputException.java:59)
    	at com.fasterxml.jackson.databind.DeserializationContext.reportInputMismatch(DeserializationContext.java:1767)
    	at com.fasterxml.jackson.databind.DeserializationContext.handleUnexpectedToken(DeserializationContext.java:1541)
    	at com.fasterxml.jackson.databind.DeserializationContext.handleUnexpectedToken(DeserializationContext.java:1446)
    	at com.fasterxml.jackson.databind.DeserializationContext.extractScalarFromObject(DeserializationContext.java:958)
    	at com.fasterxml.jackson.databind.deser.std.StdDeserializer._parseString(StdDeserializer.java:1441)
    	at com.fasterxml.jackson.databind.deser.std.StringDeserializer.deserialize(StringDeserializer.java:48)
    	at com.fasterxml.jackson.databind.deser.std.StringDeserializer.deserialize(StringDeserializer.java:11)
    	at com.fasterxml.jackson.databind.ObjectMapper._convert(ObjectMapper.java:4631)
```

### Environment

- Kestra Version: develop

--- a/core/src/main/java/io/kestra/core/models/property/Property.java
+++ b/core/src/main/java/io/kestra/core/models/property/Property.java
@@ -64,8 +64,18 @@ public class Property<T> {
                 throw new IllegalArgumentException(e);
             }
         } else {
-            expression = MAPPER.convertValue(value, String.class);
+            try {
+                expression = MAPPER.convertValue(value, String.class);
+            } catch (IllegalArgumentException e) {
+                // if it fails, try with writeValueAsString instead
+                try {
+                    expression = MAPPER.writeValueAsString(value);
+                } catch (JsonProcessingException e2) {
+                    throw new IllegalArgumentException(e2);
+                }
+            }
         }
+
         Property<V> p = new Property<>(expression);
         p.value = value;
         return p;

--- a/core/src/test/java/io/kestra/core/models/property/PropertyTest.java
+++ b/core/src/test/java/io/kestra/core/models/property/PropertyTest.java
@@ -6,6 +6,8 @@ import io.kestra.core.runners.RunContextFactory;
 import io.kestra.core.serializers.FileSerde;
 import io.kestra.core.storages.StorageInterface;
 import jakarta.inject.Inject;
+import lombok.Builder;
+import lombok.Getter;
 import org.junit.jupiter.api.Test;
 import org.slf4j.event.Level;
 import reactor.core.publisher.Flux;
@@ -221,5 +223,18 @@ class PropertyTest {
         var runContext = runContextFactory.of();
 
         assertThrows(IllegalVariableEvaluationException.class, () -> task.run(runContext));
+    }
+
+    @Test
+    void of() {
+        var prop = Property.of(TestObj.builder().key("key").value("value").build());
+        assertThat(prop, notNullValue());
+    }
+
+    @Builder
+    @Getter
+    private static class TestObj {
+        private String key;
+        private String value;
     }
 }

